impacket/krb5/gssapi.py

```diff
@@ -196,6 +238,70 @@ def GSS_Wrap(self, sessionKey, data, sequenceNumber, direction = 'init', encrypt
     def GSS_Unwrap(self, sessionKey, data, sequenceNumber, direction = 'init', encrypt=True, authData=None):
         return self.GSS_Wrap(sessionKey, data, sequenceNumber, direction, encrypt, authData)
 
+    def GSS_Wrap_LDAP(self, sessionKey, data, sequenceNumber, direction = 'init', encrypt=True):
+       
+        if encrypt:
+            data += b'\x01'
+
+        token = self.WRAP()
+        token['SGN_ALG'] = GSS_HMAC
+        token['SEAL_ALG'] = GSS_RC4
+
+        if direction == 'init':
+            token['SND_SEQ'] = struct.pack('>L', sequenceNumber) + b'\x00'*4
+        else:
+            token['SND_SEQ'] = struct.pack('>L', sequenceNumber) + b'\xff'*4
+
+        # Random confounder :)
+        token['Confounder'] = b(''.join([rand.choice(string.ascii_letters) for _ in range(8)]))
+
+        Ksign = HMAC.new(sessionKey.contents, b'signaturekey\0', MD5).digest()
+        Sgn_Cksum = MD5.new(struct.pack('<L',13) + token.getData()[:8] + token['Confounder'] + data).digest()
+
+        Klocal = bytearray()
+        from builtins import bytes
+        for n in bytes(sessionKey.contents):
+            Klocal.append( n ^ 0xF0)
+
+        Kcrypt = HMAC.new(Klocal,struct.pack('<L',0), MD5).digest()
+        Kcrypt = HMAC.new(Kcrypt,struct.pack('>L', sequenceNumber), MD5).digest()
+        
+        Sgn_Cksum = HMAC.new(Ksign, Sgn_Cksum, MD5).digest()
+
+        token['SGN_CKSUM'] = Sgn_Cksum[:8]
+
+        Kseq = HMAC.new(sessionKey.contents, struct.pack('<L',0), MD5).digest()
+        Kseq = HMAC.new(Kseq, token['SGN_CKSUM'], MD5).digest()
+
+        token['SND_SEQ'] = ARC4.new(Kseq).encrypt(token['SND_SEQ'])
+
+        if encrypt is False:
+            sspi_wrap = MechIndepToken.from_bytes(data)
+
+            wrap = self.WRAP(sspi_wrap.data[:32])
+            data = sspi_wrap.data[32:]
+            
+            Kseq = HMAC.new(sessionKey.contents, struct.pack('<L',0), MD5).digest()
+            Kseq = HMAC.new(Kseq, wrap['SGN_CKSUM'], MD5).digest()
+
+            snd_seq = ARC4.new(Kseq).encrypt(wrap['SND_SEQ'])
+
+            Kcrypt = HMAC.new(Klocal,struct.pack('<L',0), MD5).digest()
+            Kcrypt = HMAC.new(Kcrypt,snd_seq[:4], MD5).digest()
+            rc4 = ARC4.new(Kcrypt)
+            cipherText = rc4.decrypt(token['Confounder'] + data)[8:]
+            return cipherText[:-1], None
+        
+        elif encrypt is True:
+            rc4 = ARC4.new(Kcrypt)
+            token['Confounder'] = rc4.encrypt(token['Confounder'])
+            cipherText = rc4.encrypt(data)
+            finalData, cipherText = MechIndepToken(token.getData() + cipherText, KRB_OID).to_bytes()
+            return cipherText, finalData
+
+    def GSS_Unwrap_LDAP(self, sessionKey, data, sequenceNumber, direction = 'init'):
+        return self.GSS_Wrap(sessionKey, data, sequenceNumber, direction, encrypt=False)
```

Comment:
```suggestion
        return self.GSS_Wrap_LDAP(sessionKey, data, sequenceNumber, direction, encrypt=False)
```

Should call this one right? (beware if implementing the more global change of the other comment :) )

Reply:
Good catch !